OpenRA.Mods.Common/Traits/GainsExperience.cs

```diff
@@ -60,16 +63,18 @@ public GainsExperience(ActorInitializer init, GainsExperienceInfo info)
 			this.info = info;
 
 			MaxLevel = info.Conditions.Count;
-			var cost = self.Info.TraitInfo<ValuedInfo>().Cost;
-			foreach (var kv in info.Conditions)
-				nextLevel.Add(Pair.New(kv.Key * cost, kv.Value));
 
 			if (init.Contains<ExperienceInit>())
 				initialExperience = init.Get<ExperienceInit, int>();
 		}
 
 		void INotifyCreated.Created(Actor self)
 		{
+			var valued = self.Info.TraitInfoOrDefault<ValuedInfo>();
+			var requiredExperience = info.ExperienceModifier < 0 ? (valued != null ? valued.Cost : 0) : info.ExperienceModifier;
```

Comment:
This needs to default to 1 if valued is null, otherwise all keys will reduce to 0!

Reply:
Is this not intentional? From the description: `Defaults to the actor's value.`. If the actor has no value, this defaults to no value as well.

Reply:
The end result is that it breaks the trait. If this multiplier is 0 then all the experience levels defined in the yaml above are ignored and the unit will immediately rank up to its max level as soon as it receives any exp.

Reply:
Adding a `RulesetLoaded` check to throw an error if the modifier is -1 and Valued doesn't exist (similarly to how Armament handles invalid weapon references) would provide a way out of this.